### PR TITLE
Implement access control for item#new and item#create

### PR DESCRIPTION
**Pull request by zinosama**
@jhanggi

### CODE REVIEW

File: spec/controllers/items_controller_spec.rb [hunk 1]
```diff
@@ -2,21 +2,40 @@
...
+
       get :new
       expect(response).to render_template(:new)
       expect(assigns(:item)).to be_present
     end
   end
 
   describe "POST create" do
+    it "redirects when not logged in" do
+      post :create, item:{ name: "item1", price: 1.2 }
```
Comment (jhanggi) [inline spec/controllers/items_controller_spec.rb]: Minor quibble, but standard practice is put a space between the `:` the `{`.

File: spec/controllers/items_controller_spec.rb [hunk 2]
```diff
@@ -2,21 +2,40 @@
...
+      expect(response).to redirect_to(login_path)
+    end
+
     it "creates new item when valid" do
+      session[:admin_id] = "123"
+      allow(Admin).to receive(:find_by).with({ :id => "123" }).and_return(true)
```
Comment (jhanggi) [inline spec/controllers/items_controller_spec.rb]: We tend to prefer the newer hash syntax `id: 123` over the older hashrocket syntax. You'll still see it around in older codebases, but we've mostly cleaned it up using rubocop in current projects. Also, in this case, you don't need the curly braces.

**Comment by jhanggi** [comment]
Other than a couple syntactic bits, 👍

**Comment by zinosama** [comment]
@jhanggi

File: spec/features/add_item_spec.rb [hunk 3]
```diff
@@ -1,39 +1,36 @@
...
+      expect(page).to have_selector(".list li", count: 2)
+    end
+
+    it "redirects and flashes when valid" do
+      visit new_item_path
+      fill_in "item[name]", with: "item1" 
+      fill_in "item[price]", with: 1.23
+      click_on "Create Item"
+
+  		expect(current_path).to eq(items_path)
```
Comment (jhanggi) [inline spec/features/add_item_spec.rb]: There's some funky indentation going on here.

File: spec/features/add_item_spec.rb [hunk 4]
```diff
@@ -1,39 +1,36 @@
...
+      log_in_as("zino", "dasda")
+    end
+
+    it "displays error when invalid" do
+      visit new_item_path
+      fill_in "item[name]", with: ""
+      fill_in "item[price]", with: ""
+      click_on "Create Item"
+
+      expect(page).to have_selector(".list li", count: 2)
```
Comment (jhanggi) [inline spec/features/add_item_spec.rb]: Usually for something like this, you're better off looking for some text or something a little more specific.

File: spec/features/add_item_spec.rb [hunk 5]
```diff
@@ -1,39 +1,36 @@
...
+  		expect(page).to have_selector(".ui.success.message", count: 1)
+  	end
+
+  	it "should be default to not have a special date" do
+  		visit new_item_path
+  		fill_in	"item[name]", with: "item1"
+  		fill_in "item[price]", with: 1.24
+  		click_on "Create Item" 
+
+  		expect(page).to have_selector(".meta", :text => /^\$ 1.24$/, count: 1)
```
Comment (jhanggi) [inline spec/features/add_item_spec.rb]: `text` should use the `text:` syntax.

**Comment by zinosama** [comment]
@jhanggi Yeah the spacing issue was weird. Should be good now. Let me know what you think.

**Comment by jhanggi** [comment]
👍

**Pull request merged**